### PR TITLE
Fixes mob_holders not going into storage

### LINKED ISSUE #53872: you can no longer place scooped up mobs (mob_holder objs) into bags, hats, or storages
## Round ID:

local, latest master but has happened on live rounds

## Reproduction:

Scoop up a rat, try to put in chef hat, does not enter hat
Scoop up Ian, try to stash in bag, does not enter bag
Collect bees, try to hide in duffle bag, does not enter duffle bag

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

#53305 made it so clothing objects without an `icon_state` on initialize were given `ABSTRACT` flags. 

unfortunately for `mob_holder` objs, they gained their icon *after* that step of initialization.

this PR just moves the parent call to after the `icon_state` is updated, preventing them from incorrectly gaining the abstract flag.

Fixes #53872

## Why It's Good For The Game

bee bombs are back, baby

## Changelog
:cl: Melbert
fix: Fixed being unable to put scooped up mobs (ian, bees, etc) in storage containers
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
bee bombs bee bombs

File: code/modules/mob/living/inhand_holder.dm [hunk 1]
```diff
@@ -22,6 +21,7 @@
 	if(worn_slot_flags)
 		slot_flags = worn_slot_flags
 	deposit(M)
+	. = ..()
```
Comment (Rohesie) [inline code/modules/mob/living/inhand_holder.dm:24]:
```suggestion
	. = ..()
	deposit(M)
```
We want to call parent as soon as possible during initializations.

Comment (MrMelbert) [reply code/modules/mob/living/inhand_holder.dm:24]: this wouldn't actually fix the bug, because the `icon_state` is updated via the `deposit()` proc

Comment (Rohesie) [reply code/modules/mob/living/inhand_holder.dm:24]: aha, I see it now

**Comment by msgerbs** [comment]
Wow that was stupidly simple, how did I not see that?

**Pull request merged**